element_volume/volume.py

```diff
@@ -0,0 +1,261 @@
+            paramset_desc (str): description of the parameter set
+            params (dict): clustering parameters
+            paramset_idx (int, optional): Unique parameter set ID. Defaults to None.
+        """
+        if paramset_idx is None:
+            paramset_idx = (
+                dj.U().aggr(cls, n="max(paramset_idx)").fetch1("n") or 0
+            ) + 1
+
+        param_dict = {
+            "segmentation_method": segmentation_method,
+            "paramset_desc": paramset_desc,
+            "params": params,
+            "paramset_idx": paramset_idx,
+            "paramset_hash": dict_to_uuid(
+                {**params, "segmentation_method": segmentation_method}
+            ),
+        }
+        param_query = cls & {"paramset_hash": param_dict["paramset_hash"]}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1("paramset_idx")
+            if (
+                existing_paramset_idx == paramset_idx
+            ):  # If the existing set has the same paramset_idx: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    f"The specified param-set already exists"
+                    f" - with paramset_idx: {existing_paramset_idx}"
+                )
+        else:
+            if {"paramset_idx": paramset_idx} in cls.proj():
+                raise dj.DataJointError(
+                    f"The specified paramset_idx {paramset_idx} already exists,"
+                    f" please pick a different one."
+                )
+            cls.insert1(param_dict)
+
+
+@schema
+class SegmentationTask(dj.Manual):
+    definition = """
+    -> Volume
+    -> SegmentationParamset
+    ---
+    segmentation_output_dir='': varchar(255)  #  Output directory of the segmented results relative to root data directory
+    task_mode='load': enum('load', 'trigger')
+    """
+
+
+@schema
+class Segmentation(dj.Computed):
+    definition = """
+    -> SegmentationTask
+    """
+
+    class Mask(dj.Part):
+        definition = """ # A mask produced by segmentation.
+        -> master
+        mask            : smallint
+        ---
+        mask_npix       : int       # number of pixels in ROIs
+        mask_center_x   : float     # X component of the 3D mask centroid in pixel units.
+        mask_center_y   : float     # Y component of the 3D mask centroid in pixel units.
+        mask_center_z   : float     # Z component of the 3D mask centroid in pixel units.
+        mask_xpix       : longblob  # x coordinates in pixels
+        mask_ypix       : longblob  # y coordinates in pixels
+        mask_zpix       : longblob  # z coordinates in pixels
+        mask_weights    : longblob  # weights of the mask at the indices above
+        """
+
+    def make(self, key):
+        # NOTE: convert seg data to unit8 instead of uint64
```

Comment:
I may be missing something, but I am not sure what this is referring to?

Reply:
Not sure either, perhaps something related to a precursor step prior to `cellpose`